### PR TITLE
Shooter subsystem completed

### LINKED ISSUE #2: Shooter Subsystem
Currently 2 Falcon 500's in cad design, but will be reduced to 1
1 REV Robotics Servo will also be included for the hood
Limelight Sensor will also work with the shooter most likely

### PR DESCRIPTION
**Pull request by aidund**
Completed as much as it currently can be.

Closes #2

### CODE REVIEW

**Comment by aidund** [comment]
not ready

File: src/main/java/frc/robot/Constants.java [hunk 1]
```diff
@@ -13,6 +13,11 @@
 
 public final class Constants {
     public static final double stickDeadband = 0.1;
+    public static final int shooterID = 0;
+    public static final int shooterServoID = 0;
+    public static final double shooterSpin = 0.7;
+    public static final int motorStop = 0;
+    public static final double turretPower = 0.1;
```
Comment (legoguy1000) [inline src/main/java/frc/robot/Constants.java]: Move these into the consolidated section at the bottom where @owenflatman made the sub classes.

File: src/main/java/frc/robot/RobotContainer.java [hunk 2]
```diff
@@ -17,9 +20,13 @@
 public class RobotContainer {
     /* Controllers */
     private final Joystick driver = new Joystick(0);
+    private final Joystick operator = new Joystick(0);
```
Comment (legoguy1000) [inline src/main/java/frc/robot/RobotContainer.java]: This needs to be `Joystick(1)`

File: src/main/java/frc/robot/RobotContainer.java [hunk 3]
```diff
@@ -17,9 +20,13 @@
...
 
     private final SendableChooser<String> autoChooser = new SendableChooser<>();
 
+    private final Button shooterMotor = new Button(() -> Math.abs(operator.getRawAxis(XboxController.Axis.kRightTrigger.value)) > .4);
+    private final Shooter shooter = new Shooter();
+
```
Comment (legoguy1000) [inline src/main/java/frc/robot/RobotContainer.java]: Don't worry about mapping to buttons yet.

File: src/main/java/frc/robot/RobotContainer.java [hunk 4]
```diff
@@ -57,6 +64,7 @@ public RobotContainer() {
     private void configureButtonBindings() {
         /* Driver Buttons */
         zeroGyro.whenPressed(new InstantCommand(() -> swerveDrive.zeroGyro()));
+        shooterMotor.whenHeld(new ShooterRev(shooter));
```
Comment (legoguy1000) [inline src/main/java/frc/robot/RobotContainer.java]: Same

File: src/main/java/frc/robot/RobotContainer.java [hunk 5]
```diff
@@ -17,9 +20,13 @@
...
+    private final Joystick operator = new Joystick(0);
 
     private final SendableChooser<String> autoChooser = new SendableChooser<>();
 
+    private final Button shooterMotor = new Button(() -> Math.abs(operator.getRawAxis(XboxController.Axis.kRightTrigger.value)) > .4);
+    private final Shooter shooter = new Shooter();
+
     private Command autoCommand;
 
     private static final String exampleAuto = "Example Auto";
```
Comment (legoguy1000) [inline src/main/java/frc/robot/RobotContainer.java]: Get rid of the auto stuff

**Review by legoguy1000 — APPROVED** [review]

**Pull request merged**